### ISSUE TITLE
Segmentation fault when detecting on .mp4

### ISSUE THREAD

**Issue by jlevers** [AlexeyAB/darknet#3154]
When I run

`./darknet detector demo cfg/coco.data cfg/yolov3.cfg yolov3.weights /path/to/video.mp4`

the detection window pops up, everything works properly for the first 1-3 seconds of the video, and then the window closes and the program stops.

When I have `DEBUG=1` set in my `Makefile`, I get the following error:
```shell
free(): invalid next size (fast)
Aborted (core dumped)
```

I'm running with this `Makefile` config:
```make
GPU=0
CUDNN=0
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=1
LIBSO=0
ZED_CAMERA=0
```

I tried setting line 55 of demo.c to 
```c
    int dont_close_stream = 1;
```
but that didn't solve the problem.

If anyone could shed some light on what's going on here, I'd appreciate it. Thanks!

**Comment by AlexeyAB** [comment]
* Try with several mp4/avi files, do you get the same error?

* What OpenCV version do you use?

* Try to compile with LIBSO=1 DEBUG=0  and run
`LD_LIBRARY_PATH=./:$LD_LIBRARY_PATH ./uselib data/coco.names cfg/yolov3.cfg yolov3.weights test.mp4`

**Comment by jlevers** [comment]
* I've tested it on a few different files now, and it seems that it takes longer to crash on smaller video files, while it crashes much more quickly on larger ones...but yes, I still seem to get the same error. I haven't tried any really small size, long duration videos, so maybe it would be able to make it through those?

* I'm using OpenCV 3.4.6.

* When I recompile and run the command you mentioned above, it still crashes, and displays this:
```
object names loaded 
input image or video filename: 
 Video size: [3840 x 2160]
 t_write exit 
 t_network exit 
 exit_flag: detection_data.cap_frame.size = [0 x 0]
 t_cap exit 
 t_prepare exit 
 t_detect exit 
 t_draw exit 
 show detection exit
```

**Comment by AlexeyAB** [comment]
Do you use the latest version of Darknet? latest commit from https://github.com/AlexeyAB/darknet

**Comment by jlevers** [comment]
Yes, I'm up to date. I just cloned the repository yesterday, and I just pulled to make sure a moment ago.

**Comment by jlevers** [comment]
I've done some digging, and it looks like the issue is coming from here:

https://github.com/AlexeyAB/darknet/blob/cce34712f6928495f1fbc5d69332162fc23491b9/src/image_opencv.cpp#L734

I haven't been able to figure out _why_ deleting `src` would be causing a segfault, but hopefully that helps.

**Comment by rod-hendricks** [comment]
I usually get segmentation faults when the details in my cfg do not match with that of the .data file specifically with regards to the classes and the .names file. Once that is sorted out the error is gone when running inference on the models. Not sure if this is related to your issue or not.
